src/lib/api.ts

```diff
@@ -36,6 +42,8 @@ async function fetchCommonStatsWithCaching(): Promise<CommonStats> {
     deep: true,
   })
 
+  const updatedCommonStats = await updateContractNamesInCommonStats(commonStats)
```

Comment:
You should assign commonStats variable to a new updated value so that we can cache and return commonStats with contract names here

Reply:
done!